calyx-opt/src/analysis/reaching_defns.rs

```diff
@@ -636,6 +637,50 @@ fn build_reaching_def(
             en.group.borrow().name(),
         ),
         ir::Control::Empty(_) => (reach, killed),
+        ir::Control::Repeat(ir::Repeat { body, .. }) => {
+            // (XXX) If we start using this analysis, we should think more carefully
+            // about how exactly to handle repeats. This code basically just copies
+            // what while loops do (except for its dealing with cond ports).
```

Comment:
We already use this analysis so I'm not sure what the comment means. Also, let's abstract this code a bit so that we're not literally duplicating the code. Finally, for `TODO`, `XXX`, or `NOTE` comments, put your name in there so we know who wrote it: `XXX(rachit): ...`